File: src/0066-Plus-One/0066.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
 public:
  vector<int> plusOne(vector<int>& digits) {
    for (int i = digits.size() - 1; i >= 0; i--) {
      if (++digits[i] %= 10) return digits;
    }
    digits.insert(digits.begin(), 1);
    return digits;
  }
};

int main() {
  vector<int> digits = {0};
  auto result = Solution().plusOne(digits);
  return 0;
}

File: src/0066-Plus-One/0066.rs
impl Solution {
    pub fn plus_one(digits: Vec<i32>) -> Vec<i32> {
        let mut digits = digits;
        for i in (0..digits.len()).rev() {
            digits[i] = (digits[i] + 1) % 10;
            if digits[i] != 0 {
                return digits;
            }
        }
        digits.insert(0, 1);
        digits
    }
}